test/lib/ansible_test/_data/pytest/plugins/ansible_pytest_collections.py

```diff
@@ -3,12 +3,23 @@
 __metaclass__ = type
 
 import os
-import sys
 
 # set by ansible-test to a single directory, rather than a list of directories as supported by Ansible itself
 ANSIBLE_COLLECTIONS_PATH = os.path.join(os.environ['ANSIBLE_COLLECTIONS_PATH'], 'ansible_collections')
 
 
+# this monkeypatch to _pytest.pathlib.resolve_package_path fixes PEP420 resolution for collections in pytest >= 6.0.0
+# NB: this code should never run under py2
+def collection_resolve_package_path(path):
+    """Configure the Python package path so that pytest can find our collections."""
+    for parent in path.parents:
+        if str(parent) == ANSIBLE_COLLECTIONS_PATH:
+            return parent
+
+    raise Exception('File "%s" not found in collection path "%s".' % (str(path), ANSIBLE_COLLECTIONS_PATH))
```

Comment:
How about selecting a semantically better exception?
```suggestion
    raise LookupError(
        'File "{target_path!s}" not found in collection path "{coll_path!s}".'.
        format(target_path=path, coll_path=ANSIBLE_COLLECTIONS_PATH)
    )
```
P.S. `str()` is unnecessary since the specifier does that already.

Reply:
I've removed the unnecessary `str()` usage. I've kept the rest as-is since that matches the current implementation for the pytest < 6 case.